driver/normalizer/strconv.go

```diff
@@ -1,18 +1,43 @@
 package normalizer
 
 import (
+	"fmt"
 	"strconv"
 	"unicode/utf8"
 )
 
 // Functions below are copied from strconv.Unquote and strconv.Quote.
 // Original functions are unable to escape/unescape values containing
 // multiple characters since in Go single quotes represent a rune literal
+// https://github.com/golang/go/blob/65a54aef5bedbf8035a465d12ad54783fb81e957/src/strconv/quote.go#L360
 
 // unquoteSingle is the same as strconv.Unquote, but uses ' as a quote.
 func unquoteSingle(s string) (string, error) {
+	n := len(s)
```

Comment:
I think this might be clearer as:

```go
if len(s) < 2 || s[0] != s[len(s)-1] {
   return "", strconv.ErrSyntax
}
s = s[0:len(s)-1]
```

It echoes the `len` call, but at least to my eyes, the labor of pulling it out doesn't really help legibility any, and the compiler can be trusted.

Reply:
Separately: I would probably not re-use `strconv.ErrSyntax` unless the goal is to be plug-compatible with the existing strconv method for some other reason. I think we did so by default when we cloned the code, but generally it's better to avoid re-using another package's errors unless they are designed for that purpose (the main example I can think of is `io.EOF`).

Reply:
> I think this might be clearer as:
> 
> ```go
> if len(s) < 2 || s[0] != s[len(s)-1] {
>    return "", strconv.ErrSyntax
> }
> s = s[0:len(s)-1]
> ```
> 
> It echoes the `len` call, but at least to my eyes, the labor of pulling it out doesn't really help legibility any, and the compiler can be trusted.

This makes sense, but for the sake of simplifying future maintenance I would rather keep the current version, that is, as noted in comments a copy of https://golang.org/src/strconv/quote.go?s=9955:9993#L347 (sans not re-using `ErrSyntax`)

Reply:
Is our intent to preserve patches from upstream in this code? It doesn't seem like that's a good maintenance strategy. Unless we are aiming to eventually replace this with a real library function, I would not recommend trying to maintain similarity simply for its own sake.

That said: This particular example is not a crucial point. I only want to point out that the value in keeping the code similar to its origin is only really useful if we intend to keep them in sync, and that seems like a bigger issue.

Reply:
> Separately: I would probably not re-use `strconv.ErrSyntax` unless the goal is to be plug-compatible with the existing strconv method for some other reason. I think we did so by default when we cloned the code, but generally it's better to avoid re-using another package's errors unless they are designed for that purpose (the main example I can think of is `io.EOF`).

addressed in ea68ea3